src/rgw/rgw_rest_s3.h

```diff
@@ -914,42 +914,129 @@ class s3select;
 class csv_object;
 }
 
+
+class aws_response_handler
+{//TODO this class should reside on s3select submodule 
+
+private:
+  std::string sql_result;
+  struct req_state *s;//TODO will be replace by callback
+  uint32_t header_size;
+  std::unique_ptr<boost::crc_32_type> crc32;
+  RGWOp *m_rgwop;
+  std::string m_buff_header;
+  uint64_t total_bytes_returned;
+  uint64_t processed_size;
+
+  enum header_name_En
+  {
+    EVENT_TYPE,
+    CONTENT_TYPE,
+    MESSAGE_TYPE,
+    ERROR_CODE,
+    ERROR_MESSAGE
+  };
+
+  enum header_value_En
+  {
+    RECORDS,
+    OCTET_STREAM,
+    EVENT,
+    CONT,
+    PROGRESS,
+    END,
+    XML,
+    STATS,
+    ENGINE_ERROR,
+    ERROR_TYPE
+  };
+
+  const char *PAYLOAD_LINE= "\n<Payload>\n<Records>\n<Payload>\n";
+  const char *END_PAYLOAD_LINE= "\n</Payload></Records></Payload>";
+  const char *header_name_str[5] =  {":event-type", ":content-type", ":message-type",":error-code",":error-message"};
+  const char *header_value_str[10] = {"Records", "application/octet-stream", "event", "Cont", "Progress", "End", "text/xml", "Stats", "s3select-engine-error","error"};
+
+  void push_header(const char * header_name,const char* header_value);
+
+  int create_message(u_int32_t header_len);
+
+public:
+  //12 positions for header-crc
+  aws_response_handler(struct req_state *ps,RGWOp *rgwop) : sql_result("012345678901"), s(ps),m_rgwop(rgwop),total_bytes_returned{0},processed_size{0}
```

Comment:
it's a good idea to limit the use of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) in code, and `sql_result("012345678901")` feels like we're using a magic string that obscures the magic number 12. how about this?
```c++
  static constexpr size_t header_crc_size = 12; // 12 crc bytes at the beginning of each response
 public:
  aws_response_handler(struct req_state *ps, RGWOp *rgwop) : sql_result(header_crc_size, '\0'), ...
```
i also see `sql_result = "012345678901";` in a lot of the init_ functions below. are there cases where this class gets re-initialized, or are those assignments unnecessary?

if re-initialization is needed, you can use `sql_result.resize(header_crc_size);` for that